Index: src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiTrail.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiTrail.java b/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiTrail.java
--- a/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiTrail.java	(revision 56186b00b4da616cf8e1634e1f2ec7eca20324d2)
+++ b/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiTrail.java	(date 1667423541369)
@@ -63,31 +63,63 @@
         return ResponseEntity.ok(service.searchTrails(query.getQueryValue()));
     }
 
-    //TODO: ADD ERROR REPORTING
+    /**
+     * @param id unique identifier of Trail Object
+     * @return Trail Object Matching that ID
+     */
     @GetMapping("{id}")
     public ResponseEntity<Object> getTrail(@PathVariable UUID id)
     {
+        if ( service.searchTrailsByID(id).isEmpty())
+        {
+            return new ResponseEntity<>("this trail cannot be found!", HttpStatus.BAD_REQUEST);
+        }
+
         return ResponseEntity.ok(service.searchTrailsByID(id));
     }
 
-    //TODO: ADD ERROR REPORTING
+    /**
+     * @param trailName Name of trail
+     * @return all Trail objects matching that trail name
+     */
     @GetMapping("names/{trailName}")
     public ResponseEntity<Object> getTrailByName(@PathVariable String trailName)
     {
+        if ( service.searchTrailsByTrailName(trailName).isEmpty())
+        {
+            return new ResponseEntity<>("this trail cannot be found!", HttpStatus.BAD_REQUEST);
+        }
+
         return ResponseEntity.ok(service.searchTrailsByTrailName(trailName));
     }
 
-    //TODO: ADD ERROR REPORTING
+    /**
+     * @param trailSystem the name of the TrailSystem a Trail belongs to
+     * @return all Trail objects matching that trail system
+     */
     @GetMapping("systems/{trailSystem}")
     public ResponseEntity<Object> getTrailBySystem(@PathVariable String trailSystem)
     {
+        if ( service.searchTrailsByTrailSystem(trailSystem).isEmpty())
+        {
+            return new ResponseEntity<>("this trail system cannot be found!", HttpStatus.BAD_REQUEST);
+        }
+
         return ResponseEntity.ok(service.searchTrailsByTrailSystem(trailSystem));
     }
 
-    //TODO: ADD ERROR REPORTING
+    /**
+     * @param trailDifficulty rating (enum) of any trail
+     * @return all Trail objects matching that trail rating (difficulty)
+     */
     @GetMapping("ratings/{trailDifficulty}")
     public ResponseEntity<Object> getTrailByDifficulty(@PathVariable String trailDifficulty)
     {
+        if ( service.searchTrailsByTrailDifficulty(trailDifficulty).isEmpty())
+        {
+            return new ResponseEntity<>("this trail difficulty cannot be found!", HttpStatus.BAD_REQUEST);
+        }
+
         return ResponseEntity.ok(service.searchTrailsByTrailDifficulty(trailDifficulty));
     }
 
Index: src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiReview.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiReview.java b/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiReview.java
--- a/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiReview.java	(revision 56186b00b4da616cf8e1634e1f2ec7eca20324d2)
+++ b/src/main/java/com/greenriverdev/narrowsmith/fallproject/controllers/WebApiReview.java	(date 1667423541359)
@@ -84,10 +84,14 @@
         return ResponseEntity.ok(service.searchReviewsAuthorName(query.getQueryValue()));
     }
 
-    //TODO: ADD ERROR REPORTING
     @GetMapping("{trailName}")
     public ResponseEntity<Object> getTrail(@PathVariable String trailName)
     {
+        if ( service.searchReviewsByTrailName(trailName).isEmpty())
+        {
+            return new ResponseEntity<>("this trail cannot be found!", HttpStatus.BAD_REQUEST);
+        }
+
         return ResponseEntity.ok(service.searchReviewsByTrailName(trailName));
     }
 
